constraints/src/main/java/gov/nasa/jpl/aerie/constraints/json/ConstraintParsers.java

```diff
@@ -329,14 +336,25 @@ static JsonParser<WindowsFromSpans> windowsFromSpansF(final JsonParser<Expressio
             $ -> tuple(Unit.UNIT, $.expression()));
   }
 
-  static JsonParser<ForEachActivity> forEachActivityF(final JsonParser<Expression<List<Violation>>> violationListExpressionP) {
+  static JsonParser<ForEachActivitySpans> forEachActivitySpansF(final JsonParser<Expression<Spans>> spansExpressionP) {
+    return productP
+        .field("kind", literalP("ForEachActivitySpans"))
+        .field("activityType", stringP)
+        .field("alias", stringP)
+        .field("expression", spansExpressionP)
+        .map(
+            untuple((kind, actType, alias, expression) -> new ForEachActivitySpans(actType, alias, expression)),
+            $ -> tuple(Unit.UNIT, $.activityType, $.alias, $.expression));
+  }
+
+  static JsonParser<ForEachActivityViolations> forEachActivityWindowsF(final JsonParser<Expression<List<Violation>>> violationListExpressionP) {
```

Comment:
I think this should be `forEachActivityViolationsF` or `forEachActivityConstraintsF` instead of `...WindowsF`.

Reply:
ok